app/controllers/users_controller.rb

```diff
@@ -117,9 +117,19 @@ def create
       save_worked = false
     end
     if save_worked
-      @user.send_reset_password_instructions
-      flash[:success] = "Successfully created user."
-      redirect_to(users_path) && return
+      begin
+        @user.send_reset_password_instructions
+        flash[:success] = "Successfully created user."
+      rescue Net::SMTPFatalError
+        flash[:success] = "Successfully created user but reset password instructions were not sent:
+          Net::SMTPFatalError"
+      rescue StandardError => e
+        error_message = e.message
+        flash[:error] = "Failed to create user: #{error_message}"
```

Comment:
This flash error is slightly confusing to me, I've been getting error messages like localhost(25) and auth errors when SMTP is not configured and I don't think I would know it was related to my SMTP config unless I know that an email was trying to be sent. Could we change this error message to say something like `Failed to create user: incorrectly configured SMTP config : #{error_message}` or something.

Reply:
Done.